### PR TITLE
TC-181 adding eks secret class

### PR DESCRIPTION
**Pull request by Matan-Amoyal**
TC-181

### CODE REVIEW

File: src/main/groovy/hereolp/OlpCliManager.groovy [hunk 1]
```diff
@@ -1,36 +1,46 @@
...
-    OlpCliManager(pipeline, dockerCommand, hereOlpImageName, hereOlpContainerName, hereCredentialsFilePath) {
+    OlpCliManager(pipeline, hereolpCredentialsID) {
         this.pipeline = pipeline
-        this.dockerCommand = dockerCommand
-        this.hereOlpImageName = hereOlpImageName
-        this.hereOlpContainerName = hereOlpContainerName
-        this.hereCredentialsFilePath = hereCredentialsFilePath
+        this.dockerCommand = this.pipeline.env.DOCKER
+        this.imageName = 'registry.mobiwize:5000/hereolp-cli:latest'
+        this.containerName = this.pipeline.env.JOB_NAME + '-' + this.pipeline.env.BUILD_NUMBER
```
Comment (slavik57) [inline src/main/groovy/hereolp/OlpCliManager.groovy]: This container name is not descriptive enough.
Every container that will run any command will be named the same way.
So container that creates catalogs, deletes them, creates layers, manages permissions etc will be named the same

File: src/main/groovy/aws/eks/EksSecrets.groovy [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    def clusterName
+    def hereolpCredentialsFilePath
+    def files
+
+    EksSecrets(pipeline, awsCliManager, eksCliManager, clusterName) {
+        this.pipeline = pipeline
+        this.awsCliManager = awsCliManager
+        this.eksCliManager = eksCliManager
+        this.clusterName = clusterName
+        this.hereolpCredentialsFilePath = "${this.pipeline.env.WORKSPACE}/.here/credentials.properties"
```
Comment (slavik57) [inline src/main/groovy/aws/eks/EksSecrets.groovy]: This path is defined in multiple classes. Extract it to a single class called `HereOlpEnvironment` and add a function to get this path.
Use it in all the places which reference this path

File: src/main/groovy/aws/eks/EksSecrets.groovy [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        this.files = new Files(this.pipeline)
+    }
+
+    // This function get a secret name and file name and create k8s secret
+    // https://kubernetes.io/docs/concepts/configuration/secret/#creating-your-own-secrets
+    def createSecret(secretName, secretFile) {
+        try {
+            return this.awsCliManager.runAwsCommand(
```
Comment (slavik57) [inline src/main/groovy/aws/eks/EksSecrets.groovy]: Why not add a method to `EksCliManager` called `runKubectlCommand` which adds the `this.eksCliManager.getUpdateKubeconfigCommand(this.clusterName)` prefix for every command?
The `getUpdateKubeconfigCommand` line is spread all over and it is very easy to forget it

File: src/main/groovy/hereolp/OlpCliManager.groovy (hunk 1 above)
Comment (Matan-Amoyal) [reply src/main/groovy/hereolp/OlpCliManager.groovy]: I will apply the container prefix as we did with AWS classes

File: src/main/groovy/aws/eks/EksSecrets.groovy (hunk 3 above)
Comment (Matan-Amoyal) [reply src/main/groovy/aws/eks/EksSecrets.groovy]: Done.
Also with Eksctl cli

**Comment by Matan-Amoyal** [comment]
hey @slavik57 
please review :)

**Pull request merged**